core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -31,6 +33,20 @@ object Producer {
       .asJava
 
   /**
+   * The `plainSink` can be used for publishing records to Kafka topics.
+   * The `record` contains a topic name to which the record is being sent, an optional
+   * partition number, and an optional key and value.
+   */
+  def plainSink[K, V](
+    producerProvider: () => KafkaProducer[K, V],
```

Comment:
Exactly, this is common in Akka Streams. For example TCP/HTTP connection Flows are tied to an address. If you need a new one, you can create a new Flow.